[Security Assisstant] Refactor user roles in Assistant Conversation Sharing tests

enriquesanchez-elastic opened the pull request:
Updated the primary and secondary user roles for serverless environments in the Assistant Conversation Sharing test file. Removed the conditional login logic for the secondary user, simplifying the login process to always use the standard login method.

## Summary

Summarize your PR. If it involves visual changes include a screenshot or gif.


### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials
- [ ] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be allowlisted in the cloud and added to the [docker list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking changes have been approved by the breaking-change committee. The `release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was used on any tests changed
- [ ] The PR  description includes the appropriate Release Notes section, and the correct `release_note:*` label is applied per the [guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [ ] Review the [backport guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing) and apply applicable `backport:*` labels.

### Identify risks

Does this PR introduce any risks? For example, consider risks like hard to test bugs, performance regression, potential of data loss.

Describe the risk, its severity, and mitigation for each identified risk. Invite stakeholders and evaluate how to proceed before merging.

- [ ] [See some risk examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx)
- [ ] ...

enriquesanchez-elastic committed:
[Security Assisstant] Refactor user roles in Assistant Conversation Sharing tests

Updated the primary and secondary user roles for serverless environments in the Assistant Conversation Sharing test file. Removed the conditional login logic for the secondary user, simplifying the login process to always use the standard login method.

enriquesanchez-elastic committed:
[Security Assistant] Enhance Cypress tests for Assistant Conversation Sharing

- Clear input field before typing user names to ensure a clean state.
- Added explicit timeout for loading user profile options after input.
- Improved visibility check for user profile options before interaction.
- Implemented retry handling for clicking user profile options.
- Updated assertion for message user to use 'contain' for better matching of full names.

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint_all_files --no-cache --fix'

enriquesanchez-elastic committed:
[Security Assistant] Refactor input handling in shareConversationWithUser function

- Split the input clearing and typing into separate commands for clarity.
- Maintained explicit timeout for loading user profile options after input.

enriquesanchez-elastic committed:
[Security Assistant] Fix user profile seeding in shared conversations test

Ensure secondary user profile is properly initialized by visiting a page after login in the before() hook. This fixes the issue where shareConversationWithUser would fail to find the user profile option in the dropdown.

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/339867)
* Commit: 732104789e99910b47bbffa6a06c5837cc5846c1

### Failed CI Steps
* [Serverless AI Assistant - Security Solution Cypress Tests #1](https://buildkite.com/elastic/kibana-pull-request/builds/339867#019952a9-d82b-4052-b747-3b4136b9df02)

### History
* :broken_heart: [Build #339417](https://buildkite.com/elastic/kibana-pull-request/builds/339417) failed 513a16db8eb36d88a0534f3e932df852491a9f2e
* :broken_heart: [Build #339387](https://buildkite.com/elastic/kibana-pull-request/builds/339387) failed a64bd13bf942e2fa3cf4562a59b9d7be1dd03f2e
* :broken_heart: [Build #339282](https://buildkite.com/elastic/kibana-pull-request/builds/339282) failed 18226abda06082369eddb0732b4ff712d8a0f5bb

cc @enriquesanchez-elastic

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339867","number":339867,"commit":"732104789e99910b47bbffa6a06c5837cc5846c1"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339417","number":339417,"commit":"513a16db8eb36d88a0534f3e932df852491a9f2e"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339387","number":339387,"commit":"a64bd13bf942e2fa3cf4562a59b9d7be1dd03f2e"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339282","number":339282,"commit":"18226abda06082369eddb0732b4ff712d8a0f5bb"}],"number":339867}
buildkite-pr-comment-->

enriquesanchez-elastic committed:
[Security Assistant] Add user profile verification and enhance sharing functionality in tests

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/340214)
* Commit: 1fd243b1f4bd2f2cc10715a045ea4781ed9f97b5

### Failed CI Steps
* [AI Assistant - Security Solution Cypress Tests #1](https://buildkite.com/elastic/kibana-pull-request/builds/340214#019956e1-7ede-43c5-8ebd-a9d40afc87a1)

### History
* :broken_heart: [Build #339867](https://buildkite.com/elastic/kibana-pull-request/builds/339867) failed 732104789e99910b47bbffa6a06c5837cc5846c1
* :broken_heart: [Build #339417](https://buildkite.com/elastic/kibana-pull-request/builds/339417) failed 513a16db8eb36d88a0534f3e932df852491a9f2e
* :broken_heart: [Build #339387](https://buildkite.com/elastic/kibana-pull-request/builds/339387) failed a64bd13bf942e2fa3cf4562a59b9d7be1dd03f2e
* :broken_heart: [Build #339282](https://buildkite.com/elastic/kibana-pull-request/builds/339282) failed 18226abda06082369eddb0732b4ff712d8a0f5bb

cc @enriquesanchez-elastic

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/340214","number":340214,"commit":"1fd243b1f4bd2f2cc10715a045ea4781ed9f97b5"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339867","number":339867,"commit":"732104789e99910b47bbffa6a06c5837cc5846c1"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339417","number":339417,"commit":"513a16db8eb36d88a0534f3e932df852491a9f2e"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339387","number":339387,"commit":"a64bd13bf942e2fa3cf4562a59b9d7be1dd03f2e"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339282","number":339282,"commit":"18226abda06082369eddb0732b4ff712d8a0f5bb"}],"number":340214}
buildkite-pr-comment-->

enriquesanchez-elastic committed:
[Security Assistant] Update user profile handling in Assistant Conversation Sharing tests

- Refactored login process for secondary users to simplify the logic.
- Added new functions to verify and wait for user profiles, ensuring they are properly seeded for sharing functionality.
- Enhanced test setup to include checks for user profile availability before proceeding with tests.

enriquesanchez-elastic committed:
Merge branch 'main' into fix-auth-shared-conversations-tests

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint_all_files --no-cache --fix'

enriquesanchez-elastic committed:
[Security Assistant] Enhance API call for user profile verification

- Added API headers to the user profile verification request to ensure proper versioning for the elastic assistant internal API.
- Improved the structure of the request to include necessary headers for better compatibility and functionality.

enriquesanchez-elastic committed:
[Security Assistant] Clean up import statements in API calls

- Removed duplicate import of ELASTIC_HTTP_VERSION_HEADER for better code clarity.
- Ensured consistent structure in the assistant API calls file.
